packages/nextjs/src/copy-config-files.ts

```diff
@@ -5,17 +5,76 @@ const fs = require('fs')
 
 const debug = process.env.HONEYBADGER_DEBUG === 'true'
 
-async function copyErrorJs() {
-  const targetPath = path.join('pages', '_error.js')
-  const errorJsAlreadyExists = fs.existsSync(targetPath)
-  if (errorJsAlreadyExists) {
-    // Don't overwrite an existing _error.js file.
-    // Create a _error.js.bak file instead.
-    const backupPath = path.join('pages', '_error.js.bak')
+function usesTypescript() {
+  return fs.existsSync('tsconfig.json')
+}
+
+function usesPagesRouter() {
+  return fs.existsSync('pages')
+}
+
+function usesAppRouter() {
+  return fs.existsSync('app')
+}
+
+function getTargetPath(isAppRouter = false, isGlobalErrorComponent = false) {
+  if (!isAppRouter && isGlobalErrorComponent) {
+    throw new Error('invalid arguments: isGlobalErrorComponent can only be true when isAppRouter is true')
+  }
+
+  const extension = usesTypescript() ? 'tsx' : 'js'
```

Comment:
Why is it only either `.tsx` or `.js`? Are `.ts` and `.jsx` not allowed? (pardon my ignorance on this)

Reply:
It's a good question! I am following Next.js examples and these are the extensions used. So I used just these for now.